File: packages/js-dpp/lib/errors/InvalidDocumentTransitionActionError.js
const ConsensusError = require('./ConsensusError');

class InvalidDocumentTransitionActionError extends ConsensusError {
  /**
   * @param {*} action
   * @param {Object} rawDocumentTransition
   */
  constructor(action, rawDocumentTransition) {
    super(`Document transition action ${action} is not supported`);

    this.name = this.constructor.name;

    this.action = action;
    this.rawDocumentTransition = rawDocumentTransition;

    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, this.constructor);
    }
  }

  /**
   * Get action
   *
   * @return {*}
   */
  getAction() {
    return this.action;
  }

  /**
   * Get document transition
   *
   * @return {Object}
   */
  getRawDocumentTransition() {
    return this.rawDocumentTransition;
  }
}

module.exports = InvalidDocumentTransitionActionError;

File: packages/rs-dpp/src/errors/consensus/basic/document/invalid_document_transition_action_error.rs
use crate::consensus::basic::BasicError;
use crate::consensus::ConsensusError;
use crate::errors::ProtocolError;
use bincode::{Decode, Encode};
use platform_serialization_derive::{PlatformDeserialize, PlatformSerialize};
use thiserror::Error;

#[derive(
    Error, Debug, Clone, PartialEq, Eq, Encode, Decode, PlatformSerialize, PlatformDeserialize,
)]
#[error("Document transition action {} is not supported", action)]
#[platform_serialize(unversioned)]
pub struct InvalidDocumentTransitionActionError {
    /*

    DO NOT CHANGE ORDER OF FIELDS WITHOUT INTRODUCING OF NEW VERSION

    */
    action: String,
}

impl InvalidDocumentTransitionActionError {
    pub fn new(action: String) -> Self {
        Self { action }
    }

    pub fn action(&self) -> &str {
        &self.action
    }
}

impl From<InvalidDocumentTransitionActionError> for ConsensusError {
    fn from(err: InvalidDocumentTransitionActionError) -> Self {
        Self::BasicError(BasicError::InvalidDocumentTransitionActionError(err))
    }
}
